lib/Ui/Tree.elm

```diff
@@ -0,0 +1,195 @@
+module Ui.Tree exposing (..)
+
+import Css
+import Html.Styled as Styled
+import Html.Styled.Attributes exposing (css)
+import Html.Styled.Events as StyledEvents
+import List.Extra as List
+import Tree exposing (Tree(..))
+import Tree.Zipper as Zipper exposing (Zipper)
+import Ui.Color exposing (toCssColor)
+import Ui.Icon as Icon
+import Ui.Palette
+import Ui.TextStyle as TextStyle
+
+
+type alias Model a =
+    { tree : Zipper a
+    , state : State a
+    }
+
+
+type State a
+    = Focus
+    | Search String (a -> String)
```

Comment:
I thought search was only for demo? Here it seems integrated

Reply:
The search input is for demo purposes searching in the tree is part of it right now. My reasoning was that this should result in a better DX and keep similiar things in the same module. Do you forsee any issues doing it this way?